server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -444,7 +604,10 @@ public void writeTo(StreamOutput out) throws IOException {
             o.writeString(role.roleNameAbbreviation());
             o.writeBoolean(role.canContainData());
         });
-        Version.writeVersion(version, out);
+        Version.writeVersion(nodeVersion, out);
+        if (out.getTransportVersion().onOrAfter(TRANSPORT_VERSION_VERSION)) {
+            TransportVersion.writeVersion(transportVersion, out);
```

Comment:
This means that every remote node will see the same `transportVersion` property for each `DiscoveryNode`, but they will not all be using the same transport protocol version.